crates/hyperdrive-math/src/lp.rs

```diff
@@ -600,6 +621,567 @@ impl State {
+                self.get_state_after_liquidity_update(-I256::try_from(share_proceeds)?)?;
+
+            // Recalculate the present value.
+            let present_value = updated_state.calculate_present_value(current_block_timestamp)?;
+
+            // Short-circuit if we are within the minimum tolerance.
+            if self.should_short_circuit_distribute_excess_idle_share_proceeds(
+                active_lp_total_supply,
+                starting_present_value,
+                lp_total_supply,
+                present_value,
+            )? {
+                return Ok(share_proceeds);
+            }
+
+            // If the pool is net long, we can solve for the next iteration of
+            // Newton's method directly when the net curve trade is greater than
+            // or equal to the max bond amount.
+            if net_curve_trade > I256::zero() {
+                // Calculate the max bond amount.
+                let max_bond_amount = updated_state
+                    .calculate_max_sell_bonds_in_safe(self.minimum_share_reserves())?;
+
+                // If the net curve trade is greater than or equal to the max
+                // bond amount, we can solve directly for the share proceeds.
+                let net_curve_trade = FixedPoint::from(U256::try_from(net_curve_trade)?);
+                if net_curve_trade >= max_bond_amount {
+                    let share_proceeds = updated_state
+                        .calculate_distribute_excess_idle_share_proceeds_net_long_edge_case(
+                            self.share_adjustment(),
+                            self.share_reserves(),
+                            starting_present_value,
+                            active_lp_total_supply,
+                            withdrawal_shares_total_supply,
+                        )?;
+
+                    // Simulate applying the share proceeds to the reserves and
+                    // recalculate the max bond amount.
+                    //
+                    // NOTE: We are calling this with 'self' so that original
+                    // values are used with the updated value of share_proceeds.
+                    let updated_state =
+                        self.get_state_after_liquidity_update(-I256::try_from(share_proceeds)?)?;
+                    let max_bond_amount = updated_state
+                        .calculate_max_sell_bonds_in_safe(self.minimum_share_reserves())?;
+
+                    // If the max bond amount is less than or equal to the net
+                    // curve trade, then Newton's method has terminated since
+                    // proceeding to the next step would result in reaching the
+                    // same point.
+                    if max_bond_amount <= net_curve_trade {
+                        return Ok(share_proceeds);
+                    }
+                    // Otherwise, we continue to the next iteration of Newton's
+                    // method.
```

Comment:
I think we need a `continue` here. We want to skip the derivative part of the calculation and go to the next iteration of the for-loop.